testdata/all-fields2.out.diff

```diff
@@ -0,0 +1,52 @@
+  &blueprint.Blueprint{
+  	... // 6 identical fields
+  	Groups:     {{Name: "virtualization"}},
+  	Containers: {{Source: "quay.io/fedora/fedora:latest", Name: "fedora", TLSVerify: &true, LocalStorage: true}},
+  	Customizations: &blueprint.Customizations{
+  		... // 14 identical fields
+  		OpenSCAP: &{DataStream: "/usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml", ProfileID: "xccdf_org.ssgproject.content_profile_cis", Tailoring: &{Selected: {"xccdf_org.ssgproject.content_rule_md5_check"}, Unselected: {"xccdf_org.ssgproject.content_rule_grub2_password"}}},
+  		Ignition: &{FirstBoot: &{ProvisioningURL: "https://ignition.example.com/firstboot"}},
+  		Directories: []blueprint.DirectoryCustomization{
+  			{
+  				Path:          "/var/tmp/dir",
+  				User:          string("lzap"),
+  				Group:         string("lzap"),
+- 				Mode:          "755",
++ 				Mode:          "",
+  				EnsureParents: true,
+  			},
+  			{
+  				Path:          "/var/tmp/dir",
+  				User:          string("root"),
+  				Group:         string("root"),
+- 				Mode:          "755",
++ 				Mode:          "",
+  				EnsureParents: false,
+  			},
+  		},
+  		Files: []blueprint.FileCustomization{
+  			{
+  				Path:  "/var/tmp/dir/file1",
+  				User:  string("lzap"),
+  				Group: string("lzap"),
+- 				Mode:  "644",
++ 				Mode:  "",
+  				Data:  "",
+  			},
+  			{
+  				Path:  "/var/tmp/dir/file2",
+  				User:  string("root"),
+  				Group: string("root"),
+- 				Mode:  "644",
++ 				Mode:  "",
+  				Data:  "test",
+  			},
+  		},
+  		Repositories: {{Id: "fedora-cisco-openh264", GPGKeys: {"file:///etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-$releasever-$basearch"}, Metalink: "https://mirrors.fedoraproject.org/metalink?repo=fedora-cisco-ope"..., Name: "Fedora $releasever openh264 (From Cisco) - $basearch", ...}, {Id: "empty-repo", Enabled: &false, GPGCheck: &true, RepoGPGCheck: &true, ...}},
+  		FIPS:         &true,
+  		... // 5 identical fields
+  	},
+  	Distro:  "",
+  	Arch:    "",
+  	Minimal: false,
+  }
```

Comment:
If this is your first review, please note that `out.diff` is a Go struct diff format coming out of the `google-cmp` package. It is diff-like output of a round-trip conversion of UBP>BP or vice versa. It can be hard to read.